sdk/eventhubs/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/implementation/ReactorHandlerProvider.java

```diff
@@ -42,8 +43,9 @@ ConnectionHandler createConnectionHandler(String connectionId, String hostname,
             case AMQP:
                 return new ConnectionHandler(connectionId, hostname);
             case AMQP_WEB_SOCKETS:
+                return new WebSocketsConnectionHandler(connectionId, hostname);
             default:
-                throw new IllegalArgumentException(String.format(Locale.US, "This transport type '%s' is not supported yet.", transportType));
+                throw new IllegalArgumentException(String.format(Locale.US, "This transport type '%s' is not supported.", transportType));
```

Comment:
Use the ClientLogger `logAndThrow` API whenever you are throwing an exception.

Reply:
@JonathanGiles Since this is a method that returns an instance, what would I return after logging? `null`?

```java
ConnectionHandler createConnectionHandler(String connectionId, String hostname, TransportType transportType) {
        switch (transportType) {
            case AMQP:
                return new ConnectionHandler(connectionId, hostname);
            case AMQP_WEB_SOCKETS:
                return new WebSocketsConnectionHandler(connectionId, hostname);
            default:
                logger.logAndThrow(new IllegalArgumentException(String.format(Locale.US, "This transport type '%s' is not supported.", transportType)));
                return null;
        }
    }
```

Reply:
Correct. We will have a CheckStyle rule that enforces that immediately after a logger.logAndThrow call that there must be a return statement.

Reply:
I thought it was going to cause a checkstyle error. But the bug I was thinking was: PZLA_PREFER_ZERO_LENGTH_ARRAYS "null rather than an empty array"